vars/saveCache.groovy

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/groovy
+
+def call(String cache_file, String directory, Integer clean_old = 0, String exclude="") {
+    try {
+        if ( exclude.length() > 0) {
+            sh """
+            if [ ! -f "${cache_file}" ]; then
```

Comment:
Let's assume we run `saveCache('config.sh', './src', 7)` and then `saveCache('config.sh', './src', 7, 'lisk-*')` (through a replay or a change to the `Jenkinsfile`). Let's further assume the `config.sh` file wasn't changed. The next time `checkCache("config.sh")` runs we will get the files we wanted to exclude.
Maybe we could compute the cache file name based on `cache_file + exclude`?

Reply:
Then we need to pass the exclude value to the checkCache (or restoreCache) function, which is not really desired. We could add some file like `${sha1sum}-pattern` and check if it has changed, then recreate the cache. What do you think?

Reply:
Let's do that.